task-5/task-5/src/timer.ts

```diff
@@ -0,0 +1,165 @@
+export class Timer {
+  timer
+  isRunning: boolean
+  type
+  time: number
+  timeElement
+  startBtn
+  stopBtn
+  resetBtn
+  typesButton
+  focusBtn
+  shortBtn
+  longBtn
+
+  constructor(time: number) {
+    this.timer = document.querySelector<HTMLDivElement>('#app')!
+    this.isRunning = false
+    this.time = time
+    this.type = 'focus'
+
+    this.renderInitialHtmlTemplate()
+
+    this.startBtn = document.getElementsByClassName("tm__button--green")
+    this.stopBtn = document.getElementsByClassName("tm__button--red")
+    this.resetBtn = document.getElementsByClassName("tm__button--gray")
+
+    this.timeElement = document.getElementById("tm__time")
+
+    this.typesButton = document.getElementsByClassName("tm__type")
+    this.focusBtn = document.getElementById("tm__focus")
+    this.shortBtn = document.getElementById("tm__short")
+    this.longBtn = document.getElementById("tm__long")
+
+    this.registerEventListenersForInitialHtmlTemplate()
+  }
+
+  renderInitialHtmlTemplate() {
+    this.timer!.innerHTML =
+      `<div class="tm">
+      <div class="tm__types">
+        <button id="tm__focus" class="tm__type tm__type--active">Focus</button>
+        <button id="tm__short" class="tm__type">Short Break</button>
+        <button id="tm__long" class="tm__type">Long Break</button>
+      </div>
+      <div class="tm__main">
+        <div class="tm__bar">
+          <h2 id="tm__time">${this.convertSecondsToMinutesAndSeconds(this.time)}</h2>
+        </div>
+      </div>
+      <div class="tm__buttons">
+        <button class="tm__button tm__button--green">Start</button>
+        <button class="tm__button tm__button--red">Stop</button>
+        <button class="tm__button tm__button--gray">Reset</button>
+      </div>
+    </div>`
+  }
+
+  registerEventListenersForInitialHtmlTemplate() {
+
+    Array.from(this.startBtn).forEach((el) => {
+      el.addEventListener('click', this.start.bind(this))
```

Comment:
wtf is .bind doing 😄

Reply:
Without .bind, "this" returned clicked button. I found that way to access to variables I need.